micro_sam/evaluation/automatic_mask_generation.py

```diff
@@ -0,0 +1,180 @@
+import os
+import json
+import argparse
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+from glob import glob
+import imageio.v2 as imageio
+from elf.evaluation import mean_segmentation_accuracy
+
+from micro_sam.util import precompute_image_embeddings
+from micro_sam.instance_segmentation import AutomaticMaskGenerator
+
+from .inference import get_predictor
+
+
+def remove_bg_as_biggest_element(image, masks):
+    """This function removes the background by sorting the masks based on area,
+    and ignores the object with biggest area
+    """
+    instance_labels = np.zeros((image.shape[0], image.shape[1]), dtype=int)
+
+    sorted_anns = sorted(masks, key=(lambda x: x['area']), reverse=True)
+    for j, ann in enumerate(sorted_anns):
+        if j > 0:  # workaround to ignore the background masks segmented from SAM
+            m = ann['segmentation']
+            instance_labels[m] = j
+
+    return instance_labels
+
+
+def get_range_of_search_values(input_vals):
+    if isinstance(input_vals, list):
+        search_range = np.arange(input_vals[0], input_vals[1] + 0.01, 0.01)
+        search_range = [round(e, 2) for e in search_range]
+    else:
+        search_range = [input_vals]
+    return search_range
+
+
+def _grid_search(img_name, image, gt, img_save_path, amg):
+    search_range_for_iou = get_range_of_search_values([0.6, 0.9])
+    search_range_for_ss = get_range_of_search_values([0.6, 0.95])
+
+    net_list = []
+    for iou_thresh in search_range_for_iou:
+        for stability_thresh in search_range_for_ss:
+            masks = amg.generate(pred_iou_thresh=iou_thresh, stability_score_thresh=stability_thresh)
+            instance_labels = remove_bg_as_biggest_element(image, masks)
+            m_sas, sas = mean_segmentation_accuracy(instance_labels, gt, return_accuracies=True)  # type: ignore
+
+            result_dict = {
+                "cell_name": img_name,
+                "pred_iou_thresh": iou_thresh,
+                "stability_score_thresh": stability_thresh,
+                "mSA": m_sas,
+                "SA50": sas[0],
+                "SA75": sas[5]
+            }
+            tmp_df = pd.DataFrame([result_dict])
+            net_list.append(tmp_df)
+
+    img_gs_df = pd.concat(net_list)
+    img_gs_df.to_csv(img_save_path, index=False)
+
+
+def per_image_amg(args):
```

Comment:
This should all be independent of LiveCELL. Please have a look how I separated out things in
- https://github.com/computational-cell-analytics/micro-sam/blob/dev/micro_sam/evaluation/inference.py#L306 and
- https://github.com/computational-cell-analytics/micro-sam/blob/dev/micro_sam/evaluation/livecell.py#L38

The functionality here should be generic and just take `image_paths` and `gt_paths`. It also shouldn't use `argparse`, but provide a clean functional interface instead.
And you can implement livecell specific functions using the functionality from here in `livecell.py`.

Reply:
Ahha yes, makes total sense to generalise it already! Cleaned up the functionality as per your previous snippets' expectations!